Added Test for Container Handler, ServiceHandler, PodTemplateHandler, DeploymentHandler, DaemonSetHandler, EnvVarHandler, ProbeHandler

piyush1594 opened the pull request:
#1134

hrishin commented on core/src/test/java/io/fabric8/maven/core/handler/ContainerHandlerTest.java:
```diff
@@ -0,0 +1,324 @@
+
+        containers = handler.getContainers(config, images);
+
+        assertEquals(2,containers.size());
```
Can we assert other important container attributes as well? e.g. container name

piyush1594 replied:
@hrishin getImageName function is using ImageConfiguration as input to derive the image name, in ImageConfiguration class, there are only two relevant fields `name` and `registry`. ImageConfiguration class is from docker-maven-plugin. You can check the getImageName() code [here](https://github.com/fabric8io/fabric8-maven-plugin/blob/master/core/src/main/java/io/fabric8/maven/core/handler/ContainerHandler.java#L96)

piyush1594 replied:
@hrishin This has been resolved in the new commit. Thanks

piyush1594 commented:
@hrishin I added some more assertions for ContainerPart you asked and also added one-case for FQDN. Please review. Test for ServiceHandler is also added. Thanks

piyush1594 commented on core/src/main/java/io/fabric8/maven/core/handler/ProbeHandler.java:
```diff
@@ -72,22 +67,22 @@ public Probe getProbe(ProbeConfig probeConfig)  {
 
     private HTTPGetAction getHTTPGetAction(String getUrl) {
-        if (getUrl == null) {
+        if (getUrl == null || !getUrl.subSequence(0,4).toString().equalsIgnoreCase("http")) {
```
@hrishin PTAL. I divided the test cases into further smaller test cases and refactored the code.

And the point you mentioned regarding https, that is already included, the same will check both `http` and `https`.

Thanks

hrishin commented on core/src/test/java/io/fabric8/maven/core/handler/JobHandlerTest.java:
```diff
@@ -0,0 +1,172 @@
+        }
+        //asserting the exact message because
+        // it throws the same exception in case controller name is null
+        catch(IllegalArgumentException exception) {
```
Instead of try-catch use "@Test(expected=IllegalArgumentException.class) annotation"

hrishin commented:
@piyush1594 this looks good now!
Thanks!
